### PR TITLE
#2264 updating the judging rubric

### PR DESCRIPTION
**Pull request by stenington**
Opening this as a WIP PR for now. I decided to make the rubric updates without restructuring the code too much, but there are some potential refactorings to make.

In `questions.rb`, the `field` attribute is starting to feel redundant when initializing `Question` objects. Similarly, the way `score` is pulled feels slightly redundant. I also considered moving a bunch of what's essentially rubric config into a YAML file (section, idx, text), but decided to postpone doing those sorts of changes.

And then methods like `ideation_total` in the `SubmissionScore` model feel a little silly with the new naming convention, but again I went for minimal update to get everything working with the new rubric.

### CODE REVIEW

**Comment by stenington** [comment]
Incidentally, I think this PR will also fix #2321, which I encountered again while working on the rubric.

**Comment by stenington** [comment]
@shaunxp20 If you have time today to review this before your travel, I'd appreciate it. I know there's room for refactoring, but I'm particularly interested if you see something you would hesitate to accept at all.

File: db/seeds.rb [hunk 1]
```diff
@@ -233,6 +233,7 @@
     job_title: "Software Engineer",
     bio: "I am passionate about tech and empowering girls",
   )).valid?
+  Geocoding.perform(ra.account).with_save
```
Comment (shaunxp20) [inline db/seeds.rb]: I just copied this line over locally and verified that this fixes #2321! 🙌 🏅 🍰 🎉 🎈

File: app/controllers/judge/scores_controller.rb [hunk 2]
```diff
@@ -120,35 +120,34 @@ def reject_suspended_judge
 
     def score_params
       params.require(:submission_score).permit(
-        :sdg_alignment,
-        :evidence_of_problem,
-        :problem_addressed,
+        :ideation_1,
+        :ideation_2,
+        :ideation_3,
+        :ideation_4,
```
Comment (shaunxp20) [inline app/controllers/judge/scores_controller.rb]: 👍 I like these changes/the renaming in this PR! Imho, it aligns better with the scoring section/pages, and makes it clearer which section/page they belong to! 📏 🤓 📝 👍

File: app/controllers/judge/scores_controller.rb [hunk 3]
```diff
@@ -120,35 +120,34 @@ def reject_suspended_judge
...
         :ideation_comment,
         :ideation_comment_word_count,
 
-        :app_functional,
+        :technical_1,
+        :technical_2,
```
Comment (shaunxp20) [inline app/controllers/judge/scores_controller.rb]: Btw, this was the bug in #2320 (question `#2` the demo score), I like how it's a little more clear here that it's the second question on the technical section. 👍 #️⃣ 2️⃣

File: app/javascript/judge/scores/QuestionSection.vue [hunk 4]
```diff
@@ -3,7 +3,7 @@
     <div :class="solo ? 'grid__col-md-6' : 'grid__col-12'">
       <h3>{{ title }}</h3>
 
-      <p class="refer-to">Refer to the {{ referTo }}</p>
+      <slot name="section-summary" />
```
Comment (shaunxp20) [inline app/javascript/judge/scores/QuestionSection.vue]: Still getting ramped up on Vue.js, anywho this `slot` extraction looks like a nice win! 🏁 🏎

File: app/models/submission_score.rb [hunk 5]
```diff
@@ -232,9 +232,9 @@ def self.total_possible
   def self.total_possible_for(division)
     case division
     when "junior"
-      70
+      60
     when "senior"
-      90
+      80
```
Comment (shaunxp20) [inline app/models/submission_score.rb]: Oh good to know! I didn't know there was a different max score depending on division! 🌈 🌟

File: app/models/submission_score.rb [hunk 6]
```diff
@@ -341,38 +341,36 @@ def raw_total
   end
 
   def ideation_total
-    sdg_alignment +
-      evidence_of_problem +
-        problem_addressed
+    ideation_1 +
+      ideation_2 +
+        ideation_3 +
+          ideation_4
```
Comment (shaunxp20) [inline app/models/submission_score.rb]: I wonder how these ☝️ 👇 would look on one line, and if a `reduce` or some method extraction would make sense here. *shrug, just popping a little to me with all the indentation. #no-biggie #random-thoughts

File: app/views/regional_ambassador/printable_scores/show.en.html.erb [hunk 7]
```diff
@@ -91,140 +91,63 @@
...
-              <td><%= score.evidence_of_problem %></td>
-              <td></td>
-            </tr>
-
-            <tr>
-              <td>The team addresses the problem well.</td>
-              <td><%= score.problem_addressed %></td>
-              <td></td>
-            </tr>
+            <% Questions.for(score).in_section(:ideation).each do |question| %>
```
Comment (shaunxp20) [inline app/views/regional_ambassador/printable_scores/show.en.html.erb]: Nice clean up here! I was trying to think of an adjective to add to `score` or some other way to make it clearer that we're getting the questions associated to saved scored (if I'm understanding the code correctly), but nothing is coming to me at the moment. 😖 😵 ☹️

File: db/migrate/20200115211026_change_judging_rubric_fields.rb [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    add_column :submission_scores, :technical_3, :integer, default: 0
+    add_column :submission_scores, :technical_4, :integer, default: 0
+    add_column :submission_scores, :pitch_1, :integer, default: 0
+    add_column :submission_scores, :pitch_2, :integer, default: 0
+    add_column :submission_scores, :entrepreneurship_1, :integer, default: 0
+    add_column :submission_scores, :entrepreneurship_2, :integer, default: 0
+    add_column :submission_scores, :entrepreneurship_3, :integer, default: 0
+    add_column :submission_scores, :entrepreneurship_4, :integer, default: 0
+    add_column :submission_scores, :overall_1, :integer, default: 0
+    add_column :submission_scores, :overall_2, :integer, default: 0
```
Comment (shaunxp20) [inline db/migrate/20200115211026_change_judging_rubric_fields.rb]: Nice @ adding the defaults! 👍

**Comment by shaunxp20** [comment]
> @shaunxp20 If you have time today to review this before your travel, I'd appreciate it. I know there's room for refactoring, but I'm particularly interested if you see something you would hesitate to accept at all.

I did first-pass, the changes are looking good to me! 👍 😄

File: app/views/regional_ambassador/printable_scores/show.en.html.erb (hunk 7 above)
Comment (stenington) [reply app/views/regional_ambassador/printable_scores/show.en.html.erb]: Yep, you're understanding it correctly. I agree the terminology is a little murky, but don't have any suggestions either. Maybe something will arise over time as we discuss it.

File: db/migrate/20200115211026_change_judging_rubric_fields.rb (hunk 8 above)
Comment (stenington) [reply db/migrate/20200115211026_change_judging_rubric_fields.rb]: Yeah, this is actually required or tests fail 🤷‍♂

File: app/models/submission_score.rb (hunk 6 above)
Comment (stenington) [reply app/models/submission_score.rb]: Yeah, it was really tempting to do something like
```
def ideation_total
  (1..4).reduce { |sum, n| sum + public_send("ideation_#{n}) }
end
```

and possibly even auto-generate those based on the section name and question counts, but I think you end up trading off brevity for ease of understanding the model. I'm leaving it as-is here, and we could continue to talk about this sort of change later if we want.

Comment (stenington) [reply app/models/submission_score.rb]: Oh, but I will add a test testing the section totals, if we want to refactor later.

**Pull request merged**